controllers/cloud.redhat.com/providers/kafka/strimzi.go

```diff
@@ -27,27 +28,191 @@ type strimziProvider struct {
 	Config config.KafkaConfig
 }
 
-func (s *strimziProvider) configureBrokers() error {
-	clusterName := types.NamespacedName{
+func (s *strimziProvider) defaultConnectClusterName() string {
+	return fmt.Sprintf("%s-connect", s.Env.Spec.Providers.Kafka.ClusterName)
+}
+
+func (s *strimziProvider) configureKafkaCluster() error {
+	clusterNN := types.NamespacedName{
 		Namespace: getKafkaNamespace(s.Env),
 		Name:      s.Env.Spec.Providers.Kafka.ClusterName,
 	}
+	k := strimzi.Kafka{}
+	updater, err := utils.UpdateOrErr(s.Client.Get(s.Ctx, clusterNN, &k))
+	if err != nil {
+		return err
+	}
 
-	kafkaResource := strimzi.Kafka{}
+	// ensure that kafka cluster of this same name but labelled for different env does not exist
+	if envLabel, ok := k.GetLabels()["env"]; ok {
+		if envLabel != s.Env.Name {
+			return fmt.Errorf(
+				"kafka cluster named '%s' found in ns '%s' but tied to env '%s'",
+				clusterNN.Name, clusterNN.Namespace, envLabel,
+			)
+		}
+	}
+
+	// populate options from the kafka provider's KafkaClusterOptions
+	replicas := s.Env.Spec.Providers.Kafka.KafkaClusterOptions.Replicas
+	if replicas < int32(1) {
+		replicas = int32(1)
+	}
+
+	storageSize := s.Env.Spec.Providers.Kafka.KafkaClusterOptions.StorageSize
+	if storageSize == "" {
+		storageSize = "1Gi"
+	}
+
+	version := s.Env.Spec.Providers.Kafka.KafkaClusterOptions.Version
+	if version == "" {
+		version = "2.5.0"
+	}
+
+	deleteClaim := s.Env.Spec.Providers.Kafka.KafkaClusterOptions.DeleteClaim
+
+	k.Spec = &strimzi.KafkaSpec{
+		Kafka: strimzi.KafkaSpecKafka{
+			Version:  &version,
+			Replicas: replicas,
+			Listeners: []strimzi.KafkaSpecKafkaListenersElem{
+				strimzi.KafkaSpecKafkaListenersElem{Name: "tcp", Type: "internal", Tls: false, Port: 9092},
+				strimzi.KafkaSpecKafkaListenersElem{Name: "tls", Type: "internal", Tls: true, Port: 9093},
+			},
+			Storage: strimzi.KafkaSpecKafkaStorage{
+				Type:        strimzi.KafkaSpecKafkaStorageTypePersistentClaim,
+				Size:        &storageSize,
+				DeleteClaim: &deleteClaim,
```

Comment:
We already have the `pvc` option for the providers, can we piggy back off this? Is it likely that you'll want zookeeper and kafka to have different PVC policies?